File: Programmers/완전탐색/carpet.py
def solution(brown, yellow):
    answer = []
    num = 1
    yellow_pair = []
    while yellow:
        if yellow % num == 0 and num <= yellow // num:
            yellow_pair.append((num, yellow // num))
        num += 1
        if num >= yellow:
            break
    brown_pair = []
    for i in range(len(yellow_pair)):
        x, y = yellow_pair[i][0], yellow_pair[i][1]
        while x*y-yellow < brown:
            x += 2
            y += 2
        if x * y == brown + yellow:
            return [y, x]
    return answer

print(solution(24, 24))

File: Programmers_Java/완전탐색/Carpet.java
package Programmers;

public class Carpet {
    public int[] solution(int brown, int yellow) {
        int[] answer = new int[2];
        int N = brown + yellow;
        for (int c = 3; c < N / 2; c++) {
            if (N % c != 0) continue;
            int r = N / c;
            if (((2 * c) + 2 * (r - 2)) == brown && ((r - 2) * (c - 2)) == yellow) {
                answer[0] = r;
                answer[1] = c;
                break;
            }
        }
        return answer;
    }
}
